From 309c7ee3d298b43ad65cca922248a2896ee1dbcd Mon Sep 17 00:00:00 2001
From: Matti Picus <matti.picus@gmail.com>
Date: Thu, 18 Apr 2024 09:24:57 +1000
Subject: [PATCH] do not add LIBNAMESUFFIX to so

---
 exports/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/exports/Makefile b/exports/Makefile
index 27a291f34..4e54795d0 100644
--- a/exports/Makefile
+++ b/exports/Makefile
@@ -178,7 +178,7 @@ FEXTRALIB += -lm
 EXTRALIB += -lm
 else
 ifeq ($(FIXED_LIBNAME),1)
-INTERNALNAME = $(LIBPREFIX)$(LIBNAMESUFFIX).so
+INTERNALNAME = $(LIBPREFIX).so
 else
 INTERNALNAME = $(LIBPREFIX).so.$(MAJOR_VERSION)
 endif
-- 
2.39.3 (Apple Git-146)

