diff --git a/driver/level3/gemm.c b/driver/level3/gemm.c
index e37d86c28..4adfbaa39 100644
--- a/driver/level3/gemm.c
+++ b/driver/level3/gemm.c
@@ -59,16 +59,24 @@
 #define GEMM_Q 128
 #endif
 
+#define STRINGIFY2(x) #x
+#define STRINGIFY(x) STRINGIFY2(x)
+
 #ifdef GEMM_DIVIDE_RATE
 #define DIVIDE_RATE GEMM_DIVIDE_RATE
 #endif
 
 #ifdef GEMM_DIVIDE_LIMIT
 #define DIVIDE_LIMIT GEMM_DIVIDE_LIMIT
+#pragma message "GEMM_DIVIDE_LIMIT=" STRINGIFY(GEMM_DIVIDE_LIMIT)
+#else
+#pragma message "GEMM_DIVIDE_LIMIT not defined"
 #endif
 
 #ifdef THREADED_LEVEL3
+#pragma message "THREAD_LEVEL3 defined"
 #include "level3_thread.c"
 #else
+#pragma message "THREAD_LEVEL3 not defined"
 #include "level3.c"
 #endif
